File: JS/Car.js
class Car {
    constructor(license, driver) {
        this.id;
        this.license = license;
        this.driver = driver;
        this.passenger;
    }

    printDataCar() {
        console.log(this.driver)
        console.log(this.driver.name)
        console.log(this.driver.document)
    }

    set passenger(passenger) {
        if (passenger === 4) {
            this._passenger = passenger;
        } else {
            console.log("Necesitas asiganar 4 pasajeros!");
        }
    }

    get id() {
        return this._id;
    }

    set id(id) {
        this._id = id;
    }

    get license() {
        return this._license;
    }

    set license(license) {
        this._license = license;
    }

    get driver() {
        return this._driver;
    }

    set driver(driver) {
        this._driver = driver;
    }
}

File: C#/Car.cs
namespace UberPoo
{
    public class Car
    {
        private int id;
        private string license;
        private Account driver;
        protected int passenger;

        //constructor
        public Car(string license, Account driver)
        {
            this.license = license;
            this.driver = driver;
        }

        //Metodo imprimir datos del carro
        public void PrintDataCar()
        {
            if (passenger != null)
            {
                Console.WriteLine("License: " + license + " Name Driver: " + driver.name + " Pasajeros: " + passenger);
            }
        }

        public void SetPassenger(int passenger)
        {
            if (passenger == 4)
            {
                this.passenger = passenger;
            }
            else
            {
                Console.WriteLine("Necesitas asignar 4 pasajeros!");
            }
        }


        public int Passenger
        {
            get
            {
                return passenger;
            }
            set
            {
                if (value == 4)
                {
                    passenger = value;
                }
                else
                {
                    Console.WriteLine("Necesitas asignar 4 pasajeros!");
                }
            }
        }

        //Gettes and setter
        public int Id
        {
            get
            {
                return id;
            }
            set
            {
                id = value;
            }
        }

        public string License
        {
            get
            {
                return license;
            }
            set
            {
                license = value;
            }
        }

        public Account Driver
        {
            get
            {
                return driver;
            }
            set
            {
                driver = value;
            }
        }
    }

}